WalletWasabi/Services/WalletService.cs

```diff
@@ -375,28 +314,26 @@ private async Task LoadDummyMempoolAsync(SmartTransaction[] unconfirmedTransacti
 
 				lock (TransactionProcessingLock)
 				{
-					var count = 0;
-					foreach (var tx in unconfirmedTransactions)
+					foreach (var tx in BitcoinStore.TransactionStore.MempoolStore.GetTransactions())
 					{
-						if (mempoolHashes.Contains(tx.GetHash().ToString().Substring(0, compactness)))
+						uint256 hash = tx.GetHash();
+						if (mempoolHashes.Contains(hash.ToString().Substring(0, compactness)))
 						{
 							TransactionProcessor.Process(tx);
 
-							Logger.LogInfo($"Transaction was successfully tested against the backend's mempool hashes: {tx.GetHash()}.");
-							count++;
+							Logger.LogInfo($"Transaction was successfully tested against the backend's mempool hashes: {hash}.");
+						}
+						else
+						{
+							BitcoinStore.TransactionStore.MempoolStore.TryRemove(tx.GetHash(), out _);
```

Comment:
Given `TransactionStore` was designed with batch removal in mind, it would be good to consider doing this in batch instead of one by one.

Reply:
I removed the remaining batch operations. I was going with that direction at first, but they ended up being smelly.

Note, the serialization is throttled and batched.